tests/testthat/test-cut.R

```diff
@@ -0,0 +1,439 @@
+        )
+
+        expect_equivalent(
+            right_false,
+            cut(ds$birthyr,
+                3,
+                name = "new_var",
+                label = c("one", "two", "three"),
+                right = FALSE)
+        )
+
+        expect_equivalent(
+            set_breaks,
+            cut(ds$birthyr,
+                c(-1.4967, 0, 1.6662),
+                name = "new_var",
+                label = c("one", "two"),
+                right = FALSE)
+        )
+        expect_equivalent(
+            dig_lab,
+            cut(ds$birthyr,
+                3,
+                dig.lab = 2,
+                name = "new_var")
+        )
+    })
+
+    test_that("cut input validation", {
+        expect_error(
+            cut(ds$birthyr,
+                breaks = c(-1.4967),
+                name = "new_var",
+                right = TRUE),
+            "invalid number of intervals"
+        )
+
+        expect_error(
+            cut(ds$birthyr,
+                breaks = c(-1.4967, 0, 1.6662, 3),
+                name = "new_var",
+                label = c("one", "two"),
+                right = TRUE),
+            paste0("lengths of ", sQuote("breaks"), " and ", sQuote("labels"), " differ")
+        )
+        expect_error(
+            cut(ds$birthyr,
+                breaks = c(1, 1, 1),
+                name = "new_var",
+                label = c("one", "two"),
+                right = TRUE),
+            paste0(sQuote("breaks"), " are not unique")
+        )
+    })
+})
+
+with_test_authentication({
+    ds <- newDataset(df)
+
+    test_that("cut returns the same thing for crunch variables and identical vectors", {
+        ds$cat_var1 <- cut(ds$v1, 3, name = "new_var1", label = c("one", "two", "three"))
+        expect_identical(
+            cut(df$v1, 3, label = c("one", "two", "three")),
+            as.vector(ds$cat_var1)
+        )
+
+        ds$cat_var2 <- cut(ds$v1, 3, name = "new_var2", label = c("one", "two", "three"),
+            right = FALSE)
+        expect_identical(
+            cut(df$v1, 3, label = c("one", "two", "three"), right = FALSE),
+            as.vector(ds$cat_var2)
+        )
+
+        ds$cat_var3 <- cut(ds$v1,
+            c(-1.4967, 0, 1.6662),
+            name = "new_var3",
+            label = c("one", "two"),
+            right = FALSE)
+        expect_identical(
+            cut(df$v1, c(-1.4967, 0, 1.6662), label = c("one", "two"), right = FALSE),
+            as.vector(ds$cat_var3)
+        )
+
+        ds$cat_var4 <- cut(ds$v1,
+            3,
+            dig.lab = 2,
+            name = "new_var4")
+        expect_identical(
+            cut(df$v1, 3, dig.lab = 2),
+            as.vector(ds$cat_var4)
+        )
+
+        ds$cat_var5 <- cut(ds$v1,
+            3,
+            include.lowest = TRUE,
+            name = "new_var5")
+        expect_identical(
+            cut(df$v1, 3, include.lowest = TRUE),
+            as.vector(ds$cat_var5)
+        )
+    })
```

Comment:
I'd say the first use case our users will have is turning "age" or "birthyear" into something like "age4", binned like 18-29, 30-44, 45-64, 65+. A test (and man page example) illustrating that would be very helpful.